Rebase griio upstream and switch to pybind11 bindings

ekigwana opened the pull request:
Rebases griio upstream from gnuradio/gnuradio:master and switches from the now obsolete swig bindings to pybind11 bindings.

This PR requires work currently not in the target branch

1. Rebase from upstream
2. Depends on PR #4 

You are also welcome to cherry pick just the commit that [adds pybind11 bindings](https://github.com/analogdevicesinc/gnuradio/pull/5/commits/57d08e997558596edfb5456804835bc7d20fa4b2)

pwicks86 committed:
runtime: Add terminate handling

This consists of the following changes:

1. Add a top_block parameter to control exception handling

This restores past behavior where the scheduler catches exceptions
raised in block threads, allowing flowgraphs to continue running after
the failure of an individual block. It also adds optional new behavior,
selected by setting catch_exceptions=False to the top block, which causes
exceptions to not be caught. In this mode of operation, a std::terminate
handler can be installed to print a stack trace before the flowgraph exits.

2. Add terminate_handler to top_block_impl

This adds a terminate_handler function that prints information
about any uncaught exceptions and also attempts to print a backtrace
for said exception after which it calls std::abort. The backtrace is
printed via libunwind, which is a new optional dependency. If libunwind
is not found/installed, then the terminate handler will still print
information about the exception and then exit.

Co-Authored-By: Scott Torborg <storborg@gmail.com>

eblot committed:
Fix drag n' drop issue with Quartz (macOS) backend

rear1019 committed:
peak_detector2: Return correct number of consumed items

This fixes #3345.

Callyan committed:
peak_detector2: bug fix and add a test case

rear1019 committed:
peak_detector2: Make the recently added test case fail

rear1019 committed:
peak_detector2: (Hopefully) correct fix for #3345

marcusmueller committed:
GRC: Actions.py needed code formatting to be workable

marcusmueller committed:
GRC: Actions with preferences now explicitly default to something

japm48 committed:
docs: use MathJax for HTML documentation

Fixes #2707.

A full LaTeX installation is no longer required for generating the HTML
docs. Also adds some nice-to-have functionalities accessible via
right-click context menu.

Details:
 - SVG backend selected by default.
 - MathJax recommends not (ab)using their CDN, so
   this adds a local dependency.
 - Some spurious doxygen warnings were also removed.

eigenphase committed:
gr-uhd: Add command uhd handler for mtune ("manual tune")

This command implements uhd::tune_request_t completely (and not just a
subset). Also includes updates to the manual.

Signed-off-by: Martin Braun <martin@gnuradio.org>

haakov committed:
cppgen: Update to modern CMake

mormj committed:
digital: add constellation_encoder block

velichkov committed:
qa: Run qtgui tests with xvfb-run when needed

If running on Linux, the DISPLAY environment variable is not set and
xvfb-run is present then run the Qt tests with xvfb-run. This would
allows running the Qt test in the gnuradio's CI.

dkozel committed:
utils: Added print-all to gnuradio-config-info

Signed-off-by: Derek Kozel <derek@bitstovolts.com>

noc0lour committed:
cmake: use TIMESTAMP function for build date and generate constants once

Previously a a custom python command was used to generate the build date
but this can be done with native CMake functions for all supported CMake
versions.

Additionally it's compatible with reproducible builds since CMake 3.8 as
it honors SOURCE_DATE_EPOCH set in the environment varibales.

Thanks to Sebastian Koslowski for pointing out the right way in CMake.

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

jdemel committed:
uhd: Add PC clock resync command

USRPs may be synchronized to a host clock. Over time USRP and host clock
deviate. This is an issue for timed commands. With this commit it is
possible to resync both devices.

The new command key is `pc_clock_resync`. The command does not expect
any argument.

It may be possible that USRP time is not strictly monotonic anymore. If
that is a requirement, other options should be explored.

e.g. connect a 'Message Strobe' block to a USRP command port and send a
message with `pmt.dict_add(pmt.make_dict(), pmt.intern('pc_clock_resync'), pmt.PMT_T)`
periodically. It is sufficient to send such a message every couple of
seconds.

mormj committed:
digital: update description of const enc/dec remove unpacked

marcusmueller committed:
blocks: use posix_fadvise to tell OS about sequential access and prefetch

jdemel committed:
blocks: Use templates more efficiently in add_blk

Compare recent changes in the moving_average block with those templatized
`volk_add` functions instead of whole work functions being templatized.

It should now be easier to add further templates for other block types.

marcusmueller committed:
grc freedesktop: Readme XML->YML

marcusmueller committed:
GRC: add metainfo file

marcusmueller committed:
GRC / Metainfo: add unit test

Depends on appstreamcli; so only enabled when that is found.

eigenphase committed:
uhd: Add GPIO handlers for messages

This allows toggling GPIOs via messages.

Signed-off-by: Martin Braun <martin@gnuradio.org>

eigenphase committed:
uhd: Add start time capability to GRC bindings

sashko committed:
grc: Make it possible to copy error message

adamhorden commented:
Thanks for this PR @ekigwana :grimacing:. Hopefully my PR will get merged as this depends on it. I was waiting for my PR to be merged so we could move gr-iio to pybind11 bindings.

I will test this PR against some actual hardware.

ekigwana commented:
@adamhorden, I fixed the bindings for setting parameters for all modules. I test using FMCOMMS2 hardware.

Commit:
digital: Modified OFDM example to utilize the Packet Header Generator

Signed-off-by: Tim Huggins <huggins.timothy@gmail.com>

ThomasHabets committed:
digital/correlate_access_code: Prevent false triggers

Leftover data from last time CAC triggered, or just leading zeroes
being ignored.

This commit prevents these by assuring that we've processed at least
as many bits as the code is, before allowing a trigger.

ThomasHabets committed:
filter/fir_filter_with_buffer: Remove manual memory management

This removes the copy constructor from being exported to Python. Which
is good, because calling copying this Would Be Bad.

ThomasHabets committed:
filter: Remove manual memory management in QA

elms committed:
Remove hardcoded GR_PYTHON_DIR

GrPython.cmake now sets the default correctly thanks to #2939

Closes #3729

daniestevez committed:
blocks: prevent losing rx_time precision in gr_read_file_metadata

The utility gr_read_file_metadata uses parse_file_metadata.parse_header()
from gr-blocks to parse and print headers in a metadata file. Currently,
the rx_time field is printed with us precision. However, the rx_time is
stored as the integer seconds of the UNIX timestamp in an uint64_t plus
a double storing the fraction of a second, so the rx_time has more
precision.

A precision on the order of 1ns is necessary for many ranging applications,
since 1us is approximately 300m of range.

This modifies parse_file_metadata.parse_header() to print rx_time with 16
decimal places, without losing precision. Aditionally, it adds "rx_time_secs"
and "rx_time_fracs" fields to the dict() return value of parse_header(), in
case they are ever needed. The "rx_time" field in this dict() is a double,
and as such it is unable to store a UNIX timestamp with 1ns precision.

ThomasHabets committed:
Add `override` for common virtual function overrides

Mostly done with:

```
find -name "*_impl.h" | xargs sed -i -r '/(void forecast|int work|int general_work|bool check_topology)\(/{:back /\)/b nxt;N;b back;:nxt s/\)$|\)(;)/) override\1/g}'
```

Then I removed an incorrect `work` that this found.

jacobagilbert committed:
blocks: improvements to the message_debug block

aggregates the data to be printed into a stringstream so that it is printed at the same time reducing debug information becoming muddled between competing cout statements. this change also provides an argument to this block to disable the printing of PDU uniform vectors which are often not needed for debug purposes

trabucayre committed:
gnuradio-runtime, pmt: dont use CMAKE_INSTALL_PREFIX for target INSTALL_INTERFACE

Yamakaja committed:
gr-filter: Add complex band rejection design helpers

ThomasHabets committed:
dtv/atsc: Remove manual memory management

ThomasHabets committed:
dtv: Remove manual memory management

All removed except some tricky stuff in dvb_ldpc_bb_impl.

ryanvolz committed:
cmake: pybind: Use "${PYTHON_EXECUTABLE}" instead of "python3".

This fixes all of these commands on Windows and wherever "python3" is
not available. The CMake-found PYTHON_EXECUTABLE will always work or
error out earlier if python is not found.

ryanvolz committed:
cmake: Set VERSION_PATH to be only "git", fixing RC_MAINT_VERSION.

ryanvolz committed:
cmake: Fix pybind11 compiler definitions with MSVC.

-Wall is not set for MSVC, and -Wno-unused-variable doesn't exist, so
put it behind an if statement.

sdh11 committed:
grc: Remove extra logger in grc/core/platform

This is an update for PR #2424. The extra debug output from GRC was
being generated by another logger instance in grc/core/platform that
did not respect the command line arguments.

This removes the extra logger and sets the logger level in main to
whatever is requested from the command line. See grc/main.py:43 for more
detail.

ryanvolz committed:
lib, include: Include missing headers.

This fixes compilation with MSVC on conda-forge, although why this
wasn't needed earlier is a mystery.

jacobagilbert committed:
modtool: CMake template should allow building OOT modules without python if desired

The gr_modtool utility should respect the -DENABLE_PYTHON=OFF setting to allow modules to be built without python support. This also does not install any GRC files even though it is possible to use GRC to generate flowgraphs without runtime python; however, this disabling python _and_ using GRC for C++ generation is a very advanced feature so the included note should be sufficient to inform anyone using in this way.

drmpeg committed:
gr-qtgui: Fix Frequency Sink Spectrum Width selection YAML.

maitbot committed:
fix header file install

mormj committed:
modtool: leave out bindings for python only modules

ryanvolz committed:
digital: python: bindings: Use nullptr instead of nonstandard __null.

This fixes compilation with MSVC.

ryanvolz committed:
vocoder: cmake: Use pybind macros to manage bindings.

This helps prevent current and future headaches by making sure
improvements to the pybind11 CMake don't also need to be added
specifically for gr-vocoder.

ryanvolz committed:
cmake: doxygen: Use ${PYTHON_EXECUTABLE} instead of python3.

ryanvolz committed:
network: lib: Use std functions for sleep instead of usleep.

This fixes compilation with MSVC.

ryanvolz committed:
network: lib: Rename s_host variable so MSVC doesn't barf.

I don't know why, but MSVC fails with the variable name as "s_host", so
make it "str_host" instead.

ryanvolz committed:
qtgui: python: bindings: Use nullptr instead of nonstandard __null.

This fixes compilation with MSVC.

PE1NUT committed:
blocks: comment fix, typo in complex_to_real.h, complex_to_imag.h

argilo committed:
grc: remove six

argilo committed:
grc: update the "Get Involved" dialog

mbr0wn committed:
uhd: Fix gain type hiding

Gain type hiding was inverted: It was supposed to hide gain type for
default (absolute/dB), and not hide for normalized and absolute/dBm.
Fixed by this commit.

mbr0wn committed:
uhd: Fix normalized gain GRC example

Fixes an issue in uhd_normalized_gain.grc, where the gain type was
set to absolute/dB.

mbr0wn committed:
runtime: gateway: Fix Doxygen warning

Fixes an incorrectly named Doxygen parameter.

mbr0wn committed:
docs: Exclude gr-newmod from Doxygen

doronbehar committed:
gr-filter/build: don't install examples if python is disabled

ThomasHabets committed:
filter: remove needless empty destructors and problematic const

You can't move from a const object, so unfortunately it's not possible
to keep constness on these.

This should fix #3757

drmpeg committed:
gr-dtv: Repair UHD example with normalized gain.

drmpeg committed:
gr-uhd: Repair UHD examples with normalized gain.

argilo committed:
runtime: remove six

argilo committed:
digital: remove six

argilo committed:
audio: fix lcm calculation

argilo committed:
docs: remove six

argilo committed:
filter: remove the last reference to __future__

argilo committed:
gru: remove unused files

argilo committed:
runtime: remove snprintf

drmpeg committed:
gr-blocks: Add missing methods to Multiply Const block pybind11 binding.

ryanvolz committed:
fec: lib: Remove use of FEC_API for function definitions.

This fixes building with MSVC. Since these files first get built into an
object that is *not* a shared library, FEC_API was resolving to
__declspec(dllimport), which should never be used for function
definitions. Warnings still occur about inconsistent dll linkage because
the gr_fec_rs target is built into an object library that doesn't define
"gnuradio_fec_EXPORTS", but the end result works. These warnings have
always existed previously.

drmpeg committed:
gr-blocks: Add missing methods to Add Const block pybind11 binding.

mormj committed:
grc: replace deprecated getargspec with getfullargspec to remove warning

mormj committed:
digital: replace deprecated getargspec with getfullargspec to remove warning

jacobagilbert committed:
Replace custom undocumented string hashing algorithm with std::hash. The std implmentation is slightly faster and has been tested to have similar distribution with arbitrary input data

jacobagilbert committed:
Make hash table a little larger to reduce expensive iteration. The memory impact is very small, on the order of 10's of kB.

drmpeg committed:
pybind11: Remove SWIG remnants.

jacobagilbert committed:
pmt: add absolute pmt serialization QA

currently there is nothing to check the absolute accuracy of PMT serialization, and it is easy for changes to break this provided serialize_str() and deserialize_str() are inverse functions. this provides some absolute reference PMT strings to ensure compatibility is retained.

argilo committed:
runtime: remove unused HAVE_SNPRINTF macro

The last reference to HAVE_SNPRINTF was removed in 
a75f0c40fe0912535fe102792f27c13d01df8e3e so it should now be safe to 
remove it.

argilo committed:
cmake: remove check for six module

dl1ksv committed:
Adding optional parameter addl_includes to gr_modtool bind, space added

argilo committed:
vocoder: refactor get_next_tx_char callback

The get_next_tx_char callback in the FreeDV modulator block is one of 
the last places where snprintf is used. I've eliminated it by using C++ 
string methods instead. This also removes the 79-character limit.

This change also fixes a bug where the last character of the message was 
not transmitted.

Lastly, I refactored the callback to be a static member function, 
eliminating the need for a separate structure to hold the callback 
state.

argilo committed:
vocoder: add text message output to FreeDV demodulator

The FreeDV protocol includes a low bit rate text stream which is used to 
transmit information such as call sign and location. GNU Radio's FreeDV 
demodulator block sets up a callback to receive this text data, but the 
callback currently does nothing.

Here I've added an optional message output to the block. It collects 
characters until a carriage return is encountered, then outputs a 
string. I've updated the sample flow graph to print these messages using 
the Message Debug block.

drmpeg committed:
gr-qtgui: Remove unused set_title() method in QT GUI Eye Sink.

doronbehar committed:
gr-fec: Enable examples only if Python is enabled.

gr-fec's grc example files are usable only with Python.

argilo committed:
grc: fix pylint C0303: Trailing whitespace

argilo committed:
grc: fix pylint C0326: wrong number of spaces

jacobagilbert committed:
pmt: add binding for __repr__ function

per 3.8 behavior, and the GR wiki, the __repr__ function of a PMT object should be overloaded to call pmt::write_string. This is currently not bound so this adds this binding.

argilo committed:
grc: fix pylint C0330: wrong hanging indentation

boggonpi committed:
gr-blocks: Stream Mux C++ improvements

- replace C-style cast with reinterpret_cast<>
- make some variables const
- prefer in-class initializers for static data
- avoid copies in for loop by using a reference

jacobagilbert committed:
gr-uhd: fix misspelling of 'request' in several locations

dl1ksv committed:
Modify template to cast Python pointer correctly, unnecessary comments removed

mormj committed:
cmake: manual override to ignore pybind hash checking

This permits an OOT to override the hash checking

putting a 0 for BINDTOOL_HEADER_FILE_HASH will ignore the hash checking.
This would:
1) Not be the default behavior
2) only for OOTs
3) Use at your own risk

/***********************************************************************************/
/* This file is automatically generated using bindtool and can be manually edited  */
/* The following lines can be configured to regenerate this file during cmake      */
/* If manual edits are made, the following tags should be modified accordingly.    */
/* BINDTOOL_GEN_AUTOMATIC(0)                                                       */
/* BINDTOOL_USE_PYGCCXML(0)                                                        */
/* BINDTOOL_HEADER_FILE(foo.h)                                                     */
/* BINDTOOL_HEADER_FILE_HASH(0)                                                    */
/***********************************************************************************/

ryanvolz committed:
audio: lib: windows: Add <cctype> header for std::isdigit.

Previously the header must have been pulled in by one of the Boost
includes, because with a newer Boost version I had build errors. This
makes the include explicit so std::isdigit is guaranteed to be declared.

mormj committed:
modtool: update of cli block name checking

argilo committed:
vocoder: allow input & output rates to differ

The input & output rates of the FreeDV modulator aren't necessarily the 
same. In particular, the 2400A and 2400B modes have an input rate of 
8000 sps and an output rate of 48000 sps. To allow for arbitrary input & 
output rates, I've changed freedv_tx_ss from sync_block to block.

mormj committed:
modtool: check blockname before other checks

drmpeg committed:
gr-vocoder: Install missing block tree file.

japm48 committed:
digital: ofdm_equalizer_simpledfe: add enable_soft_output option

Enable outputting equalized symbols that are not exactly constellation
symbols. This is useful for OFDM soft decoding/demodulation (e.g.
optimal turbo or LDPC decoding).

For compatibility, this new behaviour has to be enabled by setting the
new option "enable_soft_output" to true (false by default).

mormj committed:
Merge pull request #3809 from argilo/digital-remove-six

digital: remove six

mormj committed:
runtime: remove six (#3808)

japm48 committed:
runtime: fix Windows logging

This removes a few compilation errors in Windows related to logging
functions (see PR #3339).

japm48 committed:
audio: fix Windows logging

This removes a few compilation errors in Windows related to logging
functions (see PR #3339).

jacobagilbert committed:
gr-uhd: adding a bool to force tune request if a 'direction' key is provided with the tune command. This fixes an issue when the 'TX' and 'RX' directions are tuned explicitly to the same frequency; prior to this fix the second tune request is dropped.

drmpeg committed:
gr-video-sdl: Fix YUV formats.

dl1ksv committed:
Make additional includes and defines available in the bind process

dl1ksv committed:
Update qa_qtgui.py

Test fails with
  File "/home/schroer/gnuradiocomponents/gnuradio-volker/gr-qtgui/python/qtgui/qa_qtgui.py", line 78, in test13
    self.qtsnk = qtgui.eye_sink_c(1024, 1, "Test", 1, None)
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. gnuradio.qtgui.qtgui_python.eye_sink_c(size: int, samp_rate: float, nconnections: int = 1, parent: gnuradio.qtgui.qtgui_python.QWidget = None)

Invoked with: 1024, 1, 'Test', 1, None

vowstar committed:
gr-uhd/lib/usrp_block_impl.cc: fix compile error using boost>=1.73

When using boost>=1.73 and compile with uhd, it always have errors.
Fix the compile error when using uhd and gcc10: error: ‘_1’ was not declared
Bug: https://bugs.gentoo.org/751259

usrp_source_impl.cc: use lambdas instead of boost::bind
usrp_block_impl.cc:use lambdas instead of boost::bind
usrp_block_impl.h: Switch from boost::function to std::function

https://github.com/gnuradio/gnuradio/pull/3441
https://github.com/gnuradio/gnuradio/pull/3887

usrp_source_impl: fix parameter bug of original code

The usrp_block_impl::register_msg_cmd_handler uses cmd_handler_t as
handler type, but it have 3 parameters.
Fix the parameter mismatch bug and change the signature as cmd_handler_t

Signed-off-by: Huang Rui <vowstar@gmail.com>

mormj committed:
modtool: minor cleanup makeyaml

MattMills committed:
zeromq: Add ZMQ_LINGER value to prevent infinite block

Closes: #1132
Per the ZMQ documentation update, the docs originally listed the default
of ZMQ_LINGER as 30 seconds, however the real default was -1.

This caused the behavior of blocking indefinitely on top_block.stop()
while the socket waited for abandoned messages to be read by a client.

Ideally this value should be configurable, I've opened #3872 as follow up.

mormj committed:
pybind: add implicit conversion from int to enum

* dtv: add implicit conversion from int to enum
* analog: add implicit conversion from int to enum
* blocks: add implicit conversion from int to enum
* digital: add implicit conversion from int to enum
* fec: add implicit conversion from int to enum
* fft: add implicit conversion from int to enum
* filter: add implicit conversion from int to enum
* qtgui: add implicit conversion from int to enum
* trellis: add implicit conversion from int to enum
* vocoder: add implicit conversion from int to enum
* bindtool: add implicitly_convertible to all enums

argilo committed:
Remove gcd, lcm, and log2 from gru module

argilo committed:
uhd: Use integer sample rates to fix decimation calculation

luzpaz committed:
Fix typos throughout the code

* Fix various typos

Found via `codespell v2.0.dev`  
`codespell -q 3 -L ans,fo,hist,inout,ist,ith,nd,sinc,uint -S ./volk`

boggonpi committed:
gr-blocks: Add example flowgraphs for Stream Mux/Demux blocks.

Add a couple of example flowgraphs for these blocks. Usage is
demonstrated in the wiki page of each block.

argilo committed:
Remove hexint and hexshort from gru module

argilo committed:
Remove freqz and gnuplot_freqz from gru module

argilo committed:
Remove gru.daemonize and dial_tone_daemon example

These were added in GNU Radio 3.1.3, but a Google search finds no 
mention of gru.daemonize outside of the changelog. Modern Python 
programs could use the python-daemon library, which implements the 
daemon specification of PEP 3143.

argilo committed:
Fix examples that use wavfile_sink

The recent libsndfile transition changed the API of the wavfile_sink 
block, so the two examples that use it need to be updated.

mormj committed:
qa: remove xml file parameter causing deprecation warnings

mormj committed:
qa: run autopep8 formatting on qa python files

find ./ -iname qa*.py | xargs autopep8 --in-place -a -a

mostly formats whitespace and gets rid of trailing semicolons

mormj committed:
qa: fix typo

marcusmueller committed:
modernization: `override` instead of virtual in all compilation units

marcusmueller committed:
add modernize-use-override to clang-tidy checks

argilo committed:
digital: remove obsolete digital voice blocks

As far as I can tell, the last reference to these classes was removed in 
2c1ac6a86dada1a16fa4d112781678c595af46b2 back in 2011. They are 
non-functional, since they haven't kept up with API changes like the 
transition to hier_block2.

argilo committed:
Remove unused gru imports

Neither of these files use the gru module anymore. I've tested that they 
work correctly after removing the unused imports.

vowstar committed:
usrp_block:pybind: add missing bind for cmd_power_key

Signed-off-by: Huang Rui <vowstar@gmail.com>

boggonpi committed:
grc: Save config before execution and update recent file sub-menu.

Saves the grc prefs before executing a flowgraph to avoid losing them if
anything goes wrong and grc needs to be forcibly closed.

Also fixes recent files sub-menu refresh. When a file is opened, opened
from recent, or saved as, the recent files sub-menu is now updated.

willcode committed:
rotator: add getter rotator::phase()

willcode committed:
freq_xlating_fir_filter: keep phase continuous on retune

mormj committed:
Move FFT functions and blocks to templates (#3903)

* fft: refactor fft classes as templates

In order to reduce code duplication construct fft objects and blocks
with template paramters.

Separate forward and reverse as a template
parameter

Maintain the real fft block forward as r2c and reverse as r2c as well

Co-authored-by: Andrej Rode <mail@andrejro.de>

mormj committed:
pmt: use to_string for char string_refs

willcode committed:
qa_freq_xlating_fir_filter: adjust phase to match code

marcusmueller committed:
analog: WFM tools: proper integer div, decimation handling

drmpeg committed:
gr-filter: Remove unused parameter in polyphase_filterbank.

mormj committed:
modtool: Change handled exception to ModuleNotFoundError

boggonpi committed:
gr-analog: Rename wfm_rcv_pll blocks

Rename from auto-generated to more sensible names.

Signed-off-by: David Pi <david.pinho@gmail.com>

drmpeg committed:
build: Remove clang unused variable warnings.

drmpeg committed:
build: Remove clang override warnings.

drmpeg committed:
build: Remove clang move assignment operator deleted warnings.

jacobagilbert committed:
grc: restore pre-3.8 block id behavior

Prior to 3.8 block IDs were shown in the properties dialogue for every block as these are important for function of some GR features. #2795 brought this back as an option for 3.8+, however it forces the 'hide' value to effectively be 'none' instead of 'part' (not visible on the block canvas but visible in the properties dialogue) for all blocks. This changes this back to the pre-3.8 behavior where the ID value is not shown on the canvas but is shown in the properties dialogue.

boggonpi committed:
grc: Fix multiple 'Apply' in modify block parameters dialog box

When the modify parameters dialog block is constructed, the instance of
the block to be modified, is passed in. When the apply button is clicked
there is an extraenous flowgraph.import_data() call that re-creates all
blocks in the flowgraph. The consequence is that the block instance
passed to the dialog box is no longer the one in the flowgraph, and so
subsequent Applies have no effect. This change simply removes this call.

boggonpi committed:
grc: Fix message port show/hide problem due to bus logic update

update_bus_logic() is called when disabling a bus, it doesn't check if
it's been previously enabled. This causes it to restore port.hidden with
a None which ultimately is converted to False. So it never hides the
message ports.
Also, filter message ports from buses.

dl1ksv committed:
Update parameter background colours realtime

marcusmueller committed:
move msgq_runner out of runtime into gr-uhd/apps, its single consumer

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
runtime: abolish the gru python module

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
digital/examples/gen_whitener: import cleanup

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
Set minimum CMake version variable to what CMake demands anyway

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
Bump the C++ standard version to C++14

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

drmpeg committed:
gnuradio-runtime: Fix qa_fast_atan2f test code.

mormj committed:
grc: remove gnome-terminal deprecation warning

When running a "No GUI" flowgraph from GRC, it gives the following
deprecation warning:
Option “-e” is deprecated and might be removed in a later version of gnome-terminal.
Use “-- ” to terminate the options and put the command line to execute after it.

This follows the suggested fix to update the deprecation

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

mormj committed:
filter: replace rational_resampler_base with rational_resampler

The bulk of the rational resampler code is named rational_resampler_base
and only in Python is the automatic designing of filters if taps are not
specified.  This pushes that code into the C++ classes (which get
binded) so rational_resampler.py can go away, removing a layer of
wrapping

Commit:
filter: remove initialization for history which is same as default

maitbot committed:
RPC fixes to allow using Thrift-0.13

Adds ability for master to build with newer Thrift versions
by checking availability of thrift include files.
(Also spelling fix: Thift -> Thrift)
And fix up logging for rpcserver_thrift class.

maitbot committed:
more qt4 qt5 updates

Have additional apps, examples, and tools use Qt5 instead of Qt4.

duggabe committed:
gr-audio: revise nperiods in gr-audio-alsa.conf

duggabe committed:
gr-audio: revise nperiods in gr-audio-alsa.conf

willcode committed:
grc: add wiki link to properties dialog docs (#3933)

So user's don't have to copy/paste the URL

duggabe committed:
gr-audio: revise alsa default nperiods

dl1ksv committed:
Add test to postinstall only , if testing is enabled

duggabe committed:
gr-qtgui: remove erroneous parameters in compass.block.yml (#3959)

dl1ksv committed:
Do not disable boost, if only the unit_test_framework is missing

willcode committed:
fir_filter: in constructor, 'decimation' is unused

dl1ksv committed:
doxygen: Increase DOT_GRAPH_MAX_NODES (#3923)

warning: Included by graph for 'docs/doxygen/pydoc_macros.h' not generated, too many nodes (584), threshold is 500. Consider increasing DOT_GRAPH_MAX_NODES.

dl1ksv committed:
qtgui: Resize the windows properly in gui sink

Commit:
Set C_STANDARD to 11 (highest valid value)

Commit:
gr-uhd/apps: import cleanup of msgq_runner

Commit:
gr-uhd: clean up import of msgq_runner in uhd_app.py

drmpeg committed:
gr-blocks: Fix file source block large file handling for Windows.

rgov committed:
cmake: Fix variable substitution in pkg-config

Variable interpolation is done with configure_file(... @ONLY), so use @VAR@ and not ${VAR}.

Fixes #3969.

Commit:
fec: remove stray pybind submodule definition for polar

efardin committed:
analog: fm: Fix orientation of L and R audio; Remove hard coded parameters

Signed-off-by: efardin <efardin@ieee.org>

mbr0wn committed:
Update the repository to reflect the requirement to sign the DCO

Going forward, we will be requiring contributors to sign the DCO. This
commit adds the DCO text itself and updates the contribution guidelines.

Signed-off-by: Martin Braun <martin@gnuradio.org>

Commit:
gr-digital: Update docs for MPSK SNR Estimator

miek committed:
grc: indent multiline connections in py flowgraph

This fixes flowgraph generation when using connection templates with
multiple lines. Previously it would only indent the first line and
subsequent lines would have no indentation, causing a python error.

Signed-off-by: Mike Walters <mike@flomp.net>

Commit:
digital: deprecate clock recovery blocks covered by symbol_sync

The pfb, msk and mm clock recovery blocks are all covered by the
symbol_sync block.  See https://www.youtube.com/watch?v=uMEfx_l5Oxk

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
filter: remove deprecated win_type duplicate in firdes

The win_type defined in firdes is marked as deprecated to be removed in
3.8, so let's go ahead and remove it and fix all the references

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
digital: update pybind hashes for mpsk_snr_est

Commit:
fft: populate window enums under fft.window

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
global replace of filter.firdes.WIN_ with fft.window.WIN_

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
global replace of firdes.WIN_ with fft.window.WIN_

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
filter: update qa tests with import fft

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
add import fft to python files that reference fft.window

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
replace fft.window.WIN_ with window.win_ and include in grc default import

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
filter: update python binding hashes

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
fft: update window python enums to fft.window.*

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

mbr0wn committed:
digital: ofdm: Fix whitespace in OFDM chanest block

Both the C++ and Python files have really weird whitespace issues,
making them hard to read. This makes the whitespace usage for consistent
with usual formatting, and fixes some Doxygen issues.

Signed-off-by: Martin Braun <martin@gnuradio.org>

eduardosm committed:
Allow to choose whether to normalize constellation average magnitude, average power or to not normalize at all.

Signed-off-by: Eduardo Sánchez Muñoz <esm@eduardosm.net>

drmpeg committed:
gr-digital: Document normalization parameter in constellation.h

Signed-off-by: Ron Economos <w6rz@comcast.net>

drmpeg committed:
gr-filter: Document fractional_bw parameter in rational_resampler.h

Signed-off-by: Ron Economos <w6rz@comcast.net>

willcode committed:
qtgui: toggle switch was reverse

Fixes #3939.
Toggle switch was left==on, now right==on.

Signed-off-by: Jeff Long <willcode4@gmail.com>

marcusmueller committed:
Bump dependency versions to match these of 3.9; might increase further

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
C++14: Use std::make_unique instead of boost::make_unique

Also, clean up includes

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
.clang-tidy: add modernize-make-unique check

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
digital: fll_band_edge_cc: make_unique instead of reset(new)

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
CMakeList: master now targets 3.10

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

Commit:
filter: update decimation after gcd calculation

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

Commit:
digital: fix bindings and yaml for constellation

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

efardin committed:
analog: fm: Add audio_rate variable and deemph_tau parameters

Signed-off-by: efardin <efardin@ieee.org>

willcode committed:
pfb.py: fix variable in static function

When create_taps() was moved to a static function, a _self reference
tagged along. Use function parameter as intended.

Signed-off-by: Jeff Long <willcode4@gmail.com>

drmpeg committed:
gr-qtgui: Fix Frequency Sink max hold for half spectrum width option.

Signed-off-by: Ron Economos <w6rz@comcast.net>

willcode committed:
gr-qtgui number sink: Honor block setting of 'avg' in GUI

GUI was overriding block parameter (setting to 0) at init.
- Use block parameter for 'avg'
- Add callback for 'avg' to yml

It is still possible to set the avg from the GUI middle click. The menu
value is not updated when avg is set from a block parameter (there is no
slot in the widget to do this). If the user creates a GRC file with a
widget (e.g., entry) to set avg, and also sets it via the menu, the
values shown will not match. However, the value can be successfully set
either way.

Signed-off-by: Jeff Long <willcode4@gmail.com>

haakov committed:
grc: Let test_generator.py know where to find blocks

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

haakov committed:
grc: Accept file pointer in load_stdlib() (converter/xml.py)

Fixes test_xml_parser.py failing in the absence of lxml

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

haakov committed:
grc: Clean up test_expr_utils.py

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

haakov committed:
qtgui: Fix default QT Chooser labels

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

haakov committed:
qtgui: Additional Qt Chooser fixes (squash this)

Signed-off-by: Håkon Vågsether <hakon.vagsether@gmail.com>

haakov committed:
qtgui: Add asserts to Qt Chooser

Signed-off-by: Håkon Vågsether <hakon.vagsether@gmail.com>

mbr0wn committed:
uhd: Fix unit test

The qa_uhd.py test has been broken for a long time. Now fixed.

Signed-off-by: Martin Braun <martin@gnuradio.org>

haakov committed:
qtgui: More QT Chooser fixes (squash this)

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

adriengit committed:
gr-blocks: fix default length with list in Stream Mux and Demux yml

Signed-off-by: Adrien Michel <adriengit@users.noreply.github.com>

ryanvolz committed:
runtime: lib: Fix GR_RUNTIME_API declaration in terminate_handler.h.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>

ryanvolz committed:
runtime: lib: Put #include <cxxabi.h> behind HAVE_LIBINUWIND.

It's only needed when libunwind is used, and always including it fails
on Windows because the header doesn't exist for MSVC.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>

ryanvolz committed:
filter: Fix polyphase_filterbank for MSVC, w/ default move constructor.

Because of the deleted copy constructor (previously implicit, now made
explicit in this commit to match fir_filter.h), polyphase_filterbank.cc
fails to compile with MSVC trying to use the fft_filter_ccf copy
constructor. Explicitly declaring a default move constructor fixes this
by allowing it to use the move constuctor instead, which is the
behavior we want. It's a mystery why MSVC requires this and other
compilers don't, but it works.

See #3892 for similar discussion and change for other filter classes.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>

haakov committed:
qtgui: Qt Chooser: Don't replace labels that aren't blank (Squash)

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

smunaut committed:
grc: Move snippets prep earlier in the file

They are going to be needed in the closeEvent() Qt stuff
and this is earlier in the file ...

Signed-off-by: Sylvain Munaut <tnt@246tNt.com>

smunaut committed:
grc: In flowgraph, use closeEvent() to stop flow graph and not aboutToQuit

So originally I introduced the aboutToQuit() stuff in the hope to
cleanly shutdown the flow graph, including the Qt widgets blocks
before the Qt stuff is torn down.

Turns out that in aboutToQuit some stuff is already invalid, especially
in OpenGL and so causes SegFault on exit.

Instead this patch modify the logic so that we stop the flow graph before
Qt even begin disabling things.

The two shutdown path that exits are :
 - The sig_handler in case we get TERM or INT signals
 - The closeEvent() of the main window

This makes sure that both path call tb.stop() tb.wait() and also
any snippet 'main_after_stop'.

Signed-off-by: Sylvain Munaut <tnt@246tNt.com>

ryanvolz committed:
cmake: python: Fix linking when runtime py interpreter != build time.

This fixes a segmentation fault with the Pybind11 bindings when trying
to load any Python module on OSX with a different interpreter than
was used at build time (relevant to at least conda and homebrew). This
fix was already in place for the prior Swig bindings, but the Pybind11
changes do not incorporate it since they link with ${PYTHON_LIBRARIES}
instead of the (fixed) Python::Python target.

For clearer separation and to adopt the approach used by the new-ish
(3.12) builtin CMake FindPython module, this commit creates a separate
Python::Module target to be used when creating a Python extension module
(which includes "-undefined" and "dynamic_lookup" for macOS instead of
linking directly to the Python library). The Python::Python target
that does link to the Python library is still necessary for use as a
PRIVATE target (e.g. for runtime and qtgui).

For future reference and a general description of best practices for
linking Python extension modules, see
https://blog.tim-smith.us/2015/09/python-extension-modules-os-x.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>

nickoe committed:
gr-digital: Fixup GUI hints for example_corr_est

The tab control were missing a proper GUI hint and was just placed in
the bottom corner.

Add GUI hint to it and adjust the amount of rows the other time sink
consumes to let the sliders not consume that much screen real estate.

Slightly align blocks to look nicer.

Signed-off-by: Nick Østergaard <oe.nick@gmail.com>

nickoe committed:
gr-digital: Change threshold method of example_corr_est

In example_corr_est switch to dynamic threshold instead of absolute.
This make the example do something at start up instead of just
flatlining and it seems like the example becomes more interesting.

Signed-off-by: Nick Østergaard <oe.nick@gmail.com>

ZackerySpytz committed:
gr-blocks: Check fseek() for failure

Some fseek() calls were not being checked for failure.

Signed-off-by: Zackery Spytz <zspytz@gmail.com>

guruofquality committed:
runtime: added missing algorithm header

tag_checker.h uses std::sort

Signed-off-by: Josh Blum <josh@joshknows.com>

guruofquality committed:
trellis: added missing include for string header

fsm.h uses std::string

Signed-off-by: Josh Blum <josh@joshknows.com>

willcode committed:
gr-blocks qa for peak_detector2: fix result type in test5

Signed-off-by: Jeff Long <willcode4@gmail.com>

willcode committed:
qtgui:Added Time Raster Sink Control Axis Configuration

This update allows a time raster sink's axis minimum, maximum,
and label to now be specified in the block configuration options.
This allows the block to be formatted to match a waterfall plot
in terms of having frequency on the lower axis.

Based on PR from ghostop14 <ghostop14@gmail.com>

Signed-off-by: Jeff Long <willcode4@gmail.com>

haakov committed:
grc: Don't draw disabled and hidden blocks, even if they are selected

Signed-off-by: Håkon Vågsether <hauk142@gmail.com>

nickoe committed:
gr-digital: Fix typo in digital_modulate_vector block

There appears to be an errant space between mod and .to_basic_block().
This doesn't seem to affect the operation. This makes sense as in both
python and c++ the . is an operator that can have whitespace aorund it.

This is a cosmetic fix.

Fixes #4036

Signed-off-by: Nick Østergaard <oe.nick@gmail.com>

drmpeg committed:
cmake: Fixup BUILD_DATE commit.

Signed-off-by: Ron Economos <w6rz@comcast.net>

willcode committed:
File Source: handle EOF for non-seekable files

Based on original PR by esqt.

Signed-off-by: Jeff Long <willcode4@gmail.com>

willcode committed:
gr-fft: Log Power FFT Python fix

Block would fail as soon as it was placed in GRC.
A couple of imports/renames were needed.

Signed-off-by: Jeff Long <willcode4@gmail.com>

mbr0wn committed:
fixup! uhd: Add PC clock resync command

Signed-off-by: Martin Braun <martin@gnuradio.org>

mbr0wn committed:
Merge pull request #4099 from gnuradio/fix-master-pc-sync

fixup! uhd: Add PC clock resync command

ArtemPisarenko committed:
runtime: integrate generation of math sine_table.h into build process

File sine_table.h is generated artifact and shouldn't be part of sources
in order to properly be in sync with changes to gen_sine_table.py

Signed-off-by: Artem Pisarenko <artem.k.pisarenko@gmail.com>

ArtemPisarenko committed:
runtime: simplify math fxpt sine/cosine calculation

Algorithm improved in order to eliminate getting output values
outside of range [-1.0, 1.0)

Fixes #2993

Signed-off-by: Artem Pisarenko <artem.k.pisarenko@gmail.com>

ArtemPisarenko committed:
analog: fix AGC classes QA test to match updated sig_source output

Correcting reference values in test data evaluated against graph
involving recently corrected fxpt sine/cosine calculation

Signed-off-by: Artem Pisarenko <artem.k.pisarenko@gmail.com>

marcusmueller committed:
Changelog: prepare for 3.10.0.0

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
Set C++ standard to C++17

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
clang-tidy: add modernize-deprecated-header

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

marcusmueller committed:
clang-tidy: run full .clang-tidy on C++17 codebase

run

/usr/share/clang/run-clang-tidy.py -checks=file '-header-filter=.*'
-fix ..

from build directory.

Then,

clang-format -i $(git diff --name-only origin/master)

to clang-format changed files.

Then, refresh all header hashes in pybind bindings
(*/python/bindings/*.cc)

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

mbr0wn committed:
Add GitHub Actions for master branch

For now, this is identical to the 3.9 GH actions. Should the 3.10 (aka
master) dependencies change, we need new containers here.

Signed-off-by: Martin Braun <martin@gnuradio.org>

marcusmueller committed:
Pybind: Adding header hash hotfixer

DO NOT USE THIS TOOL IF ADDING OR REMOVING ANY FUNCTIONALITY FROM A
HEADER.

This tool allows to quickly update the hash in a binding file
(*/python/*/binding/something.cc).

WARNING: This script updates header hashes without checking whether the
underlying functionality changed. If you added or removed any python-bound
function in the header to this binding file THIS IS NOT THE TOOL YOU SHOULD BE
USING. Use bindtool instead to generate correct binding code.

Signed-off-by: Marcus Müller <mmueller@gnuradio.org>

ZackerySpytz committed:
gr-qtgui: Fix deprecated use of QPixmap::grabWidget()

Use QWidget::grab() instead (which was introduced in Qt 5).

Signed-off-by: Zackery Spytz <zspytz@gmail.com>

ryanvolz committed:
runtime: lib: Don't use hard-coded absolute path constants.

This replaces the path constants for the prefix, system configuration,
and preferences directories with a runtime prefix lookup based on the
gnuradio-runtime library location and relative paths from that. Boost >=
1.61 is required, but this is now compatible with the minimum version
required by GNU Radio overall.

Also use cmake-style paths instead of native paths for substitution to
avoid unescaped character problems. Windows is able to use the
forward-slash paths just fine, and this also has external benefits (e.g.
conda will do path replacement during installation and it substitutes
forward-slash paths, so this avoids mixed-slash paths).

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>

Commit:
digital: change ofdm_chanest qa test to have noise vector

Signed-off-by: Josh Morman <jmorman@perspectalabs.com>

mbr0wn committed:
GitHub Actions: Also check .cc.in files

These files are templates that get expanded to C++ during CMake time.
They also need to follow our .clang-format, so let's check for them
during CI.

Signed-off-by: Martin Braun <martin@gnuradio.org>

mbr0wn committed:
lib: Clang-format constants.cc.in

Because we always used globs to find the files we re-format, this
slipped under the clang-format-radar.

Signed-off-by: Martin Braun <martin@gnuradio.org>

drmpeg committed:
build: Remove clang compiler warnings.

Signed-off-by: Ron Economos <w6rz@comcast.net>

ekigwana committed:
Merge remote-tracking branch 'upstream/master'

ekigwana committed:
Merge remote-tracking branch 'origin/master' into merge-griio-upstream
